### DISCUSSION TITLE
Extend Joomla 3 support for 1 more year: security fixes and PHP 7 only

### DISCUSSION THREAD

**Discussion by stevejburge** [joomla/joomla-cms#39658] [General] [answered]
I may regret sticking my head over the parapet and posting this, but here goes ...

It may be sensible to consider one more year of very limited support for Joomla 3, perhaps until August 2024.

As part of the ecosystem around Joomla, we working hard to update all our extensions for Joomla 4. But for us and others, it's a slow process. And we talk with a lot of end-users who are not able to move to Joomla 4 yet.

So I wonder if there’s a sweet spot that produces happy Joomla 3 users, eases the transition to Joomla 4/5, and produces minimal work for the core team:

- 1 more year only
- Security fixes only
- PHP 7 only

**Comment by brianteeman** [comment]
It would be useful to know why those users are unable to upgrade. In many of the cases I have seen it is because they are using old extensions that have been dead for years but still give the impression they are alive. For those users it doesn't matter how long j3 support is extended they will never be able to simply upgrade.

**Reply by PhilippSPOTTERON** [reply]
K2 is not ready yet.

**Comment by stevejburge** [comment]
Some of it is our fault, honestly. We underestimated how long it would take to upgrade all our extensions.

Some of it is a matter of time. Most extensions that will move to Joomla 4 will be upgraded by the middle of 2023. But that leaves a short window for the end-users to upgrade.

**Reply by brianteeman** [reply]
So to be clear what you are really saying is please extend the support period as we fcked up and didn't start early enough during the 3 year dev period to be ready in time and are about 18 months late. As a result we might lose our paying customers to competing products so please can you help our company.

**Reply by brianteeman** [reply]
Something else to consider is that you are not just asking for Joomla to continue to support v3 but you are also asking every template and extension developer to continue to support v3. Is that practical

**Comment by stevejburge** [comment]
Update: I do regret sticking my head over the parapet.

This is why I don't engage in Joomla any more.

**Reply by brianteeman** [reply]
what did you expect - warm fuzzy replies just because you asked

**Reply by htmgarcia** [reply]
> what did you expect - warm fuzzy replies just because you asked

@brianteeman you can at least try to be polite when answering.

**Reply by brianteeman** [reply]
I was https://github.com/joomla/joomla-cms/discussions/39658#discussioncomment-4706748

Until the real reason was revealed.

**Reply by rdeutz** [reply]
Whatever the reason is, it doesn't give the right to be not polite. If you can't say it in a nice way, stay away from saying it.

**Reply by ssnobben** [reply]
> Update: I do regret sticking my head over the parapet.
> 
> This is why I don't engage in Joomla any more.

hmm you just lost 3 license LMS purchases and this made me head for other Joomla developer. Joomla is larger than me you or just one person. Its not kindergarten.

**Comment by alantamb** [comment]
Well, to be honest I agree with Steve. 
And... no... FYI I don't a have a company that is asking for help.

**Reply by brianteeman** [reply]
Please explain the reasoning to help understand

**Comment by rdeutz** [comment, marked as answer]
Hi Steve,

we discussed this a couple of times and we are aware about all problems siteowners, extensions developers and others have. We have made ajustments with Joomla 5 based on the feedback and have found a compromise. As always not everyone is happy with it. 

One of the important changes we made over the last years was that we have timed based releases. We did very well on this and users of Joomla! can now be sure when we will make a release.

That's said and with knowing that in August Joomla 3 sites are not magically don't work anymore we don't want to extend the support for Joomla3. 

I hope the message will reach you behind the parapet.

Best Robert

**Reply by stevejburge** [reply]
Thanks @rdeutz. I appreciate you listening and taking the time to reply in detail.

Understood. We'll keep pushing onwards with our Joomla 4 updates  >>>
